napari_skimage_regionprops/_table.py

```diff
@@ -71,49 +78,15 @@ def _clicked_table(self):
-            if selected_column is not None:
-                if isinstance(self._layer, napari.layers.Labels):
-                    from ._parametric_images import visualize_measurement_on_labels, map_measurements_on_labels
-                    new_layer = self._viewer.add_image( map_measurements_on_labels(self._layer, selected_column, self._viewer) ,
-                                                        name=selected_column + " in " + self._layer.name ,
-                                                        affine=self._layer.affine ,
-                                                        scale=self._layer.scale,
-                                                        rotate=self._layer.rotate
-                                                        )
-                    new_layer.contrast_limits = [np.min(self._table[selected_column]), np.max(self._table[selected_column])]
-                    new_layer.colormap = "jet"
-                elif isinstance(self._layer, napari.layers.Points):
-                    features = self._layer.features
-                    new_layer = self._viewer.add_points(self._layer.data,
-                                                        features=features,
-                                                        face_color=selected_column,
-                                                        face_colormap="jet",
-                                                        size=self._layer.size,
-                                                        name=selected_column + " in " + self._layer.name,
-                                                        affine=self._layer.affine,
-                                                        scale=self._layer.scale,
-                                                        rotate=self._layer.rotate
-                                                        )
-                    new_layer.contrast_limits = [np.min(self._table[selected_column]), np.max(self._table[selected_column])]
-
-        if "vertex_index" in self._table.keys():
-            selected_column = list(self._table.keys())[self._view.currentColumn()]
-            print("Selected column (T)", selected_column)
-            if selected_column is not None and isinstance(self._layer, napari.layers.Surface):
-                values = np.asarray(self._table[selected_column])
-                data = self._layer.data
-                data = [np.asarray(data[0]).copy(), np.asarray(data[1]).copy(), values]
-
-                new_layer = self._viewer.add_surface(data,
-                    name=selected_column + " in " + self._layer.name)
-                new_layer.contrast_limits = [np.min(self._table[selected_column]), np.max(self._table[selected_column])]
-                if "annotation" in selected_column or "CLUSTER_ID" in selected_column:
-                    new_layer.colormap = "hsv"
-                else:
-                    new_layer.colormap = "jet"
```

Comment:
Are you sure, you can remove this entire code? It seems that this breaks backwards compatibility. E.g. where is the `new_layer.colormap` going?

Reply:
I moved the setting of the colormap into the new `create_feature_map` function. The colormaps for the respective different layers are set at the following locations:

- [Here](https://github.com/jo-mueller/napari-skimage-regionprops/blob/64fc49fa5e29dd395359839dde6ddc58e9262be6/napari_skimage_regionprops/_parametric_images.py#L42C9-L42C42) for labels layers (which will be returned as `ImageData`)
- [Here](https://github.com/jo-mueller/napari-skimage-regionprops/blob/64fc49fa5e29dd395359839dde6ddc58e9262be6/napari_skimage_regionprops/_parametric_images.py#L49) for `PointsData`
- [Here](https://github.com/jo-mueller/napari-skimage-regionprops/blob/64fc49fa5e29dd395359839dde6ddc58e9262be6/napari_skimage_regionprops/_parametric_images.py#L57) for `VectorsData`
- [Here](https://github.com/jo-mueller/napari-skimage-regionprops/blob/64fc49fa5e29dd395359839dde6ddc58e9262be6/napari_skimage_regionprops/_parametric_images.py#L74) for `SurfaceData`